store/tikv/mock-tikv/mvcc_leveldb.go

```diff
@@ -395,10 +395,98 @@ func (mvcc *MVCCLevelDB) Scan(startKey, endKey []byte, limit int, startTS uint64
 	return pairs
 }
 
-// ReverseScan implements the MVCCStore interface.
+// ReverseScan implements the MVCCStore interface. The search range is [startKey, endKey).
 func (mvcc *MVCCLevelDB) ReverseScan(startKey, endKey []byte, limit int, startTS uint64, isoLevel kvrpcpb.IsolationLevel) []Pair {
-	// TODO: Implement it.
-	panic("not implemented")
+	mvcc.mu.RLock()
+	defer mvcc.mu.RUnlock()
+
+	var mvccEnd []byte
+	if len(endKey) != 0 {
+		mvccEnd = mvccEncode(endKey, lockVer)
+	}
+	iter := mvcc.db.NewIterator(&util.Range{
+		Limit: mvccEnd,
+	}, nil)
+	defer iter.Release()
+
+	succ := iter.Last()
+	currKey, _, _ := mvccDecode(iter.Key())
+	helper := reverseScanHelper{
+		startTS:  startTS,
+		isoLevel: isoLevel,
+		currKey:  currKey,
+	}
+
+	for succ && len(helper.pairs) < limit {
+		key, ver, err := mvccDecode(iter.Key())
+		if err != nil {
+			break
+		}
+		if bytes.Compare(key, startKey) < 0 {
+			break
+		}
+
+		if ver == lockVer {
+			var lock mvccLock
+			err = lock.UnmarshalBinary(iter.Value())
+			helper.addLock(key, &lock)
+		} else {
+			var value mvccValue
+			err = value.UnmarshalBinary(iter.Value())
+			helper.addKV(key, ver, value)
+		}
+		succ = iter.Prev()
+	}
+	if len(helper.pairs) < limit {
+		helper.finishEntry(nil)
+	}
+	return helper.pairs
+}
+
+type reverseScanHelper struct {
+	startTS  uint64
+	isoLevel kvrpcpb.IsolationLevel
+	currKey  []byte
+	entry    mvccEntry
+	pairs    []Pair
+}
+
+func (helper *reverseScanHelper) addLock(key []byte, lock *mvccLock) {
+	if bytes.Compare(key, helper.currKey) != 0 {
+		helper.finishEntry(key)
```

Comment:
This function does more than its name suggests.
I think move this logic to line 428 is better.